control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -163,6 +163,30 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
+	// Reconcile global configurationn validation status
+	{
+		newCondition := metav1.Condition{
+			Type:               string(hyperv1.InvalidConfiguration),
+			Status:             metav1.ConditionTrue,
+			Reason:             "ValidationFailed",
+			ObservedGeneration: hostedControlPlane.Generation,
+		}
+		if err := config.ValidateGlobalConfig(ctx, hostedControlPlane); err != nil {
+			newCondition.Message = err.Error()
+			meta.SetStatusCondition(&hostedControlPlane.Status.Conditions, newCondition)
+		} else {
+			condition := meta.FindStatusCondition(hostedControlPlane.Status.Conditions, string(hyperv1.InvalidConfiguration))
+			// Only update condition if it exists and its status is true
+			if condition != nil && condition.Status == metav1.ConditionTrue {
```

Comment:
why do we need to `Only update condition if it exists and its status is true`?
Can't we just blindly set it to false or remove it?

Reply:
As I mentioned above, the thinking was that the condition would only show up if something was wrong. However, I'll change it to always show up.